cli/bpmetadata/tfconfig.go

```diff
@@ -522,6 +522,25 @@ func mergeExistingConnections(newInterfaces, existingInterfaces *BlueprintInterf
 	}
 }
 
+// mergeExistingOutputTypes merges existing output types from an old BlueprintInterface into a new one,
+// preserving manually authored types.
+func mergeExistingOutputTypes(newInterfaces, existingInterfaces *BlueprintInterface) {
+	if existingInterfaces == nil {
+		return // Nothing to merge if existingInterfaces is nil
+	}
+
+	existingOutputs := make(map[string]*BlueprintOutput)
+	for _, output := range existingInterfaces.Outputs {
+		existingOutputs[output.Name] = output
+	}
+
+	for i, output := range newInterfaces.Outputs {
+		if existingOutput, ok := existingOutputs[output.Name]; ok && existingOutput.Type != nil {
+			newInterfaces.Outputs[i].Type = existingOutput.Type
```

Comment:
This will overwrite type info if the output-type was generated.. In effect this keeps the existing type info  even if the output type has changed.
Let's add a condition to only update type info when it is not present in the newInterfaces. Let's also add a test case for this.

Reply:
Good catch! Modified accordingly.